### ISSUE TITLE
Data augmentation not improving performance

### ISSUE THREAD

**Issue by simenkf** [divamgupta/image-segmentation-keras#191]
I have a custom dataset with images of corrosion damages. Corresponding masks are binary, i.e. pixel values 0 of 1. Performance, as measured by IoU (frequency, mean or per class) on a test set (and also on validation) is always decreased when applying data augmentation. Even simple flipping of images reduce IoU.

Training is performed on 426 images. Test set contains 60 images (and validation 120).

Inspecting the augmented images,  by adding the below lines in the data_loader generator function, reveales rather suspect segmentation masks, as attached below. Why are these multi-colored when the original masks are binary? It looks as if one channel is flipped while another is not?

```
i = 1
if do_augment: 
    im, seg[:, :, 0] = augment_seg(im, seg[:, :, 0], augmentation_name)  
    # Following three liens added to save augmented images and masks
    cv2.imwrite("frame" + str(i) + ".png", im) 
    cv2.imwrite("seg" + str(i) + ".png", seg[:, :, 0]*255) 
    i += 1
```
![seg141](https://user-images.githubusercontent.com/31414643/78697369-fc6a3080-7900-11ea-96f1-ed68c5e75c68.png)
![seg151](https://user-images.githubusercontent.com/31414643/78697376-ff652100-7900-11ea-92c3-fab5a93d32db.png)
![seg161](https://user-images.githubusercontent.com/31414643/78697378-fffdb780-7900-11ea-8e32-d76a4858476e.png)

**Comment by simenkf** [comment]
I think I figured out the weired augmentations. Updating to non-depricated methods from ImgAug, and also performing augmentation of images and masks "simultaneously" rather than first converting the augmenter to a deterministic augmenter and then augment each in succession, I now get correct outputs from the augmentation (see below). Has still yet to evaluate the performance now with working augmentation.

Also, the repo should probably be updated too with methods recommended in the ImgAug docs. Just a few (3-5) lines that need changes, see code below. Up date and much simpler, win-win. Would you like a PR for that, @divamgupta ?

##### Original implementation  ##### 
```
# Create a deterministic augmentation from the random one 
aug_det = IMAGE_AUGMENTATION_SEQUENCE.to_deterministic() 
# Augment the input image 
image_aug = aug_det.augment_image(img)

segmap = ia.SegmentationMapsOnImage(seg, shape=img.shape)
segmap_aug = aug_det.augment_segmentation_maps(segmap)
segmap_aug = segmap_aug.get_arr_int() 

return image_aug, segmap_aug
```
##### My implementation  #####
```
segmap = ia.SegmentationMapsOnImage(seg, shape=img.shape) # note the s compared to above
image_aug, segmap_aug = IMAGE_AUGMENTATION_SEQUENCE(image=img, segmentation_maps=segmap)
segmap_aug = segmap_aug.get_arr()

return image_aug, segmap_aug
```

**Comment by divamgupta** [comment]
Hi,
Sure, a PR would be great. Thanks
